### PR TITLE
add new recipe for cpc2

### PR DESCRIPTION
**Pull request by mthang**
Describe your pull request here
* Add CPC2 recipe
----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by mthang** [comment]
@BiocondaBot please add label

File: recipes/cpc2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    - setuptools
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - make
+  host:
+    - setuptools 
+    - pip
+    - python
+    - biopython
+    - six
```
Comment (daler) [inline recipes/cpc2/meta.yaml:29]: Do you need everything here in host? Biopython and six at least are probably only runtime deps. See https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#host.

**Review by daler — CHANGES_REQUESTED** [review]

File: recipes/cpc2/meta.yaml (hunk 1 above)
Comment (daler) [reply recipes/cpc2/meta.yaml:29]: They are needed in the `run:` section, but probably not the `host:` section

File: recipes/cpc2/build.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+cp data/* ${PREFIX}/data/
+
+mkdir -p ${PREFIX}/libs/libsvm
+chmod a+x libs/libsvm/*
+cp libs/libsvm/* ${PREFIX}/libs/libsvm
+
+cd ${PREFIX}/libs/libsvm
+gzip -dc libsvm-3.18.tar.gz | tar xf -
+cd libsvm-3.18
+make clean && make
```
Comment (daler) [inline recipes/cpc2/build.sh:19]: [libsvm](https://anaconda.org/conda-forge/libsvm) should be a dependency if possible, not separately compiled within this recipe

File: recipes/cpc2/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    - make
+  host:
+    - setuptools 
+    - pip
+    - python
+    - numpy
+  run:
+    - biopython
+    - six
+    - python
```
Comment (daler) [inline recipes/cpc2/meta.yaml:32]: Looking closer, this does not appear to be a python package (in the sense of having a setup.py), so setuptools and pip are not needed either. You may need `libsvm` in `host:` though.

```suggestion
  run:
    - biopython
    - six
    - python
    - numpy
```

**Review by daler — CHANGES_REQUESTED** [review]

File: recipes/cpc2/meta.yaml (hunk 3 above)
Comment (mthang) [reply recipes/cpc2/meta.yaml]: I just removed the setuptools and pip from the yaml file. Regarding the libsvm, this cpc2 tool failed to run if the libsvm wasn't compiled in the build.sh script.

File: recipes/cpc2/build.sh (hunk 2 above)
Comment (mthang) [reply recipes/cpc2/build.sh:19]: Regarding the libsvm, this cpc2 tool failed to run if the libsvm wasn't compiled in the build.sh script.

**Review by dpryan79 — APPROVED** [review]

**Review by dpryan79 — APPROVED** [review]

**Pull request merged**